openapi-cli/src/main/java/io/ballerina/openapi/cmd/Sanitize.java

```diff
@@ -0,0 +1,314 @@
+    private void generateSanitizedOpenAPI() {
+        String openAPIFileContent;
+        try {
+            openAPIFileContent = Files.readString(Path.of(inputPath));
+        } catch (Exception e) {
+            errorStream.println(ERROR_OCCURRED_WHILE_READING_THE_INPUT_FILE);
+            exitError();
+            return;
+        }
+
+        Optional<OpenAPI> openAPIOptional = getSanitizedOpenAPI(openAPIFileContent);
+        if (openAPIOptional.isEmpty()) {
+            exitError();
+            return;
+        }
+        writeSanitizedOpenAPIFile(openAPIOptional.get());
+    }
+
+    private Optional<OpenAPI> getSanitizedOpenAPI(String openAPIFileContent) {
+        // Read the contents of the file with default parser options
+        // Sanitizing will be done after filtering the operations
+        SwaggerParseResult parserResult = new OpenAPIParser().readContents(openAPIFileContent, null,
+                new ParseOptions());
+        if (!parserResult.getMessages().isEmpty() &&
+                parserResult.getMessages().contains(UNSUPPORTED_OPENAPI_VERSION_PARSER_MESSAGE)) {
+            errorStream.println(ERROR_UNSUPPORTED_OPENAPI_VERSION);
+            return Optional.empty();
+        }
+
+        OpenAPI openAPI = parserResult.getOpenAPI();
+        if (Objects.isNull(openAPI)) {
+            errorStream.println(ERROR_OCCURRED_WHILE_PARSING_THE_INPUT_OPENAPI_FILE);
+            if (!parserResult.getMessages().isEmpty()) {
+                errorStream.println(FOUND_PARSER_DIAGNOSTICS);
+                parserResult.getMessages().forEach(errorStream::println);
+            }
+            return Optional.empty();
+        }
+
+        filterOpenAPIOperations(openAPI);
+
+        // Flatten the OpenAPI definition with `flattenComposedSchemas: true` and `camelCaseFlattenNaming: true`
+        InlineModelResolver inlineModelResolver = new InlineModelResolver(true, true);
+        inlineModelResolver.flatten(openAPI);
+
+        return sanitizeOpenAPI(openAPI);
+    }
+
+    private void writeSanitizedOpenAPIFile(OpenAPI openAPI) {
+        String outputFileNameWithExt = getOutputFileName();
+        try {
+            CodegenUtils.writeFile(targetPath.resolve(outputFileNameWithExt),
+                    outputFileNameWithExt.endsWith(JSON_EXTENSION) ? Json.pretty(openAPI) : Yaml.pretty(openAPI));
+            infoStream.printf(INFO_OUTPUT_WRITTEN_MSG, targetPath.resolve(outputFileNameWithExt));
+        } catch (IOException exception) {
+            errorStream.println(ERROR_OCCURRED_WHILE_WRITING_THE_OUTPUT_OPENAPI_FILE);
+            exitError();
+        }
+    }
+
+    private String getOutputFileName() {
+        return resolveContractFileName(targetPath, fileName + getFileExtension(), format.equals("json"));
+    }
+
+    private String getFileExtension() {
+        return (Objects.nonNull(format) && format.equals("json")) ? JSON_EXTENSION : YAML_EXTENSION;
+    }
+
+    private void filterOpenAPIOperations(OpenAPI openAPI) {
```

Comment:
Can we reuse the method `filterOpenAPIOperations` for this ?

Reply:
Yes, there are common utils between this and the flatten sub command implementation. Will reuse them when merging one with another

Reply:
Fixed via [0c67edd](https://github.com/ballerina-platform/openapi-tools/pull/1770/commits/0c67edd6cfb2855ca539beb0e06a9300d88d3fb8)